src/runner.rs

```diff
@@ -26,18 +46,42 @@ pub async fn run_test_command(
         Ok(result) => result,
         Err(e) => {
             log::error!("Error waiting for execution: {}", e);
-            return Ok(RunResult {
-                exit_code: EXIT_FAILURE,
-                failures: Vec::new(),
-            });
+            return Ok(EXIT_FAILURE);
         }
     };
-    let mut failures = Vec::<Test>::new();
-    if !result.success() {
-        let mut file_counter = FileSetCounter::default();
-        let file_sets = match output_paths
+    Ok(result.code().unwrap_or(EXIT_FAILURE))
```

Comment:
This can still benefit from a `map_or_else` and a `unwrap_or` like I mentioned previously:
https://github.com/trunk-io/analytics-cli/pull/77#discussion_r1706266311

Reply:
Yeah haven't gone through all the changes yet